security/medusa/l2/kobject_cstrmem.c

```diff
@@ -67,53 +65,48 @@ int __init cstrmem_kobject_init(void)
 /* After this is called, and returns 0, cstrmem_kobject_rmmod should be. */
 static int __exit cstrmem_kobject_unload_check(void)
 {
-	if (med_unlink_kclass(&MED_KCLASSOF(cstrmem_kobject)) != MED_ALLOW)
+	if (MED_UNLINK_KCLASS(cstrmem_kobject) != 0)
 		return -EBUSY;
 	return 0;
 }
 
 void __exit cstrmem_kobject_rmmod(void)
 {
-	med_unregister_kclass(&MED_KCLASSOF(cstrmem_kobject));
+	MED_UNREGISTER_KCLASS(cstrmem_kobject);
 }
 
 module_init(cstrmem_kobject_init);
 module_exit(cstrmem_kobject_rmmod);
+
 /* quoting the comment in module.h:
  *
  * There are dual licensed components, but when running with Linux it is the
- * GPL that is relevant so this is a non issue. 
+ * GPL that is relevant so this is a non issue.
  */
 MODULE_LICENSE("GPL");
 
-/* implementation */
-
-// static struct cstrmem_kobject storage;
-
-static struct medusa_kobject_s * cstrmem_fetch(struct medusa_kobject_s * key_obj)
+static struct medusa_kobject_s *cstrmem_fetch(struct medusa_kobject_s *key_obj)
 {
 	int i, ret;
-	struct task_struct * p;
+	struct task_struct *p;
 	struct cstrmem_kobject* kobj = (struct cstrmem_kobject*) key_obj;
 
 	rcu_read_lock();
-	//p = find_task_by_pid(storage.pid);
 	p = pid_task(find_vpid(kobj->pid), PIDTYPE_PID);
 	if (p) {
 		get_task_struct(p);
 		rcu_read_unlock();
 		ret = access_process_vm(p, (unsigned long)kobj->address, kobj->data, kobj->size, 0);
-		/* TODO: here it should count characters until the first #0 found in (0,size) boundary */
+
+		/* TODO: refactor to something more .. obvious
+		 * TODO: here it should count characters until the first #0 found in (0,size) boundary */
 		for (i = 0; (i < kobj->size) && (((char*)kobj->data)[i]); i++);
```

Comment:
https://www.kernel.org/doc/htmldocs/kernel-api/API-strnlen.html

Reply:
I don't want to try to fix it as part of this PR, since it was out of scope (I just wanted to make the codebase a little bit cleaner when I had to change there something.

I added your comment there for reference :)